design/common/multi-net-multus.md

```diff
@@ -0,0 +1,110 @@
+`hostNetwork` or not. The API will be used to define networks resource for
+Rook clusters. It enables more fine-grained control over network access.
+
+## Legacy Implementation
+
+To achieve non-flat networking model, Rook can choose to enable `hostNetwork`
+and expose host network interfaces to Storage Provider pods.
+
+Ceph Rook cluster network definition example:
+```yaml
+network:
+  hostNetwork: true
+```
+
+The Ceph operator without specifying this configuration will always
+default to pod networking.
+
+## Multi-homed Network Design
+
+Rook operators can define storage cluster's network using network provider.
+Network provider example includes host, and multus. To configure the cluster
+network, cluster CRD needs to tell the network provider the appropriate
+`NetworkInterfaceSelector`. `NetworkInterfaceSelector` will be provided as list
+of `interfaces` key-value.
+
+```yaml
+network:
+  provider: <network-provider>
+  interfaces:
+    <key>: <network-interface-selector>
+    <key>: <network-interface-selector>
+```
+
+### Network Provider
+
+Network provider determines multi-homing method and the network interface
+selector. Using host network provider, pod can use host's network namespace
+and its interfaces to use as cluster network. Using popular multi-plugin CNI
+plugin such as [multus][multus-cni], Rook operators can attach multiple network
+interfaces to pods.
+
+One thing to remember, leaving the network configuration empty will default to
+kubernetes cluster default networking.
+
+### Keys
+
+The network interface selectors with key values `public` and
+`cluster` are specified for Ceph.
+
+### Network Interface Selector
+
+Network interface selector is fed to network provider to connect pod to cluster
+network. This selector may vary from one network provider to another. For
+example, host network provider only needs to know the interface name.
+
+On the other hand, multi-plugin CNI plugin needs to know the network attachment
+definition's name and vice versa. Multi-plugin such as multus may seem to follow
+_Network Attachment Selection Annotation_ documented by Kubernetes Network Plumbing Working Group's
+[Multi-Net Spec].
+
+Any future multi-homed network provider that implements the Network Plumbing WG's Multi-net spec may
+use the `multus` network. Previously-identified providers [CNI-Genie] and [Knitter] have since gone
+dormant (upd. 2023), and Multus continues to be the most prominent known provider.
+
+## Multi-Net validation tester
+
+Rook will implement a user-runnable routine that validates a multi-net configuration based on the
+[Multi-Net Spec]. Users can input one or both of the Network Attachment Definitions (NADs) for
+Ceph's `multus` network, and the routine will perform cursory validation of the ability of the
+specified networks to support Ceph.
+
+The routine will start up **a single** web server with the specified networks attached (via NADs).
```

Comment:
Is there a more detailed flow of what the test executes, what resources are created, and what results are returned that can be analyzed and troubleshooted? Perhaps a bulleted list would help understand what is happening with the state machine.

Reply:
I'd rather the design not specify too many of the internal technical details like the state machine implementation, nginx is used as the web server with a configmap. The important design details for the test are just that there is one and only one web server and N clients per node get created, which is captured here.

Reply:
I guess I just like bullet points to describe flows, the flow is described here already, nevermind.